sys/random/shaxprng.c

```diff
@@ -124,13 +167,13 @@ void random_init(uint32_t seed)
 uint32_t random_uint32(void)
 {
     uint32_t ret;
-    int8_t bytes[sizeof(uint32_t)];
-    _random_bytes((uint8_t *)bytes, sizeof(uint32_t));
+    uint8_t bytes[sizeof(uint32_t)];
+    _random_bytes(bytes, sizeof(bytes));
 
-    ret = ((bytes[0] & 0xff) << 24)
-        | ((bytes[1] & 0xff) << 16)
-        | ((bytes[2] & 0xff) <<  8)
-        |  (bytes[3] & 0xff);
+    ret = ((uint32_t)(bytes[0] & 0xff) << 24)
+        | ((uint32_t)(bytes[1] & 0xff) << 16)
+        | ((uint32_t)(bytes[2] & 0xff) <<  8)
+        | ((uint32_t)(bytes[3] & 0xff));
```

Comment:
Reminds me a lot of https://github.com/RIOT-OS/RIOT/blob/ab88a96a4a837f64316cc26f72c340ff632af3f2/core/include/byteorder.h#L495-L498

Does it make sense to use it here or would that be overkill?

Reply:
Maybe doesn't make that much sense, since it would pull in unnecessary dependencies. But I guess the `& 0xff` are not really needed.

Reply:
> But I guess the `& 0xff` are not really needed.

Then again, you just took that from what was there before. So your call if you want to adapt that.

Reply:
(if you do, please squash immediately)